lib/rules/a11y-no-generic-link-text.js

```diff
@@ -0,0 +1,68 @@
+const {elementType, getProp, getPropValue} = require('jsx-ast-utils')
+
+const bannedLinkText = ['read more', 'here', 'click here', 'learn more', 'more', 'here']
+
+/* Downcase and strip extra whitespaces and punctuation */
+const stripAndDowncaseText = text => {
+  return text
+    .toLowerCase()
+    .replace(/[.,/#!$%^&*;:{}=\-_`~()]/g, '')
+    .replace(/\s{2,}/g, ' ')
+    .trim()
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'disallow generic link text',
+      url: require('../url')(module)
+    },
+    schema: []
+  },
+
+  create(context) {
+    return {
+      JSXOpeningElement: node => {
+        if (elementType(node) !== 'a') return
```

Comment:
In general we will probably use the `Link` component from `@primer/react` more often than raw anchor tags, and this check would miss those.

I don't think we can catch all of the possible cases just using a linter, but it might be more thorough to apply this check to any element with an `href` prop. That will at least catch the cases where we have components that are thin wrappers around anchor tags.

Reply:
I have a follow up PR to address this. I'd love your feedback there -- https://github.com/github/eslint-plugin-github/pull/283.